### PR TITLE
add Voltage Extension class

### PR DESCRIPTION
**Pull request by EtienneLt**
Signed-off-by: Etienne Lesot <etienne.lesot@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

it adds a new extension for the limitviolations with voltage contraints. It is similar to Current Extension but for voltage.

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.security.LimitViolation;
+
+/**
+ * Extension to handle pre-contingency voltage value for a voltage limit violation
+ *
+ * @author Olivier Bretteville <olivier.bretteville at rte-france.com>
+ */
+public class VoltageExtension implements Extension<LimitViolation> {
```
Comment (mathbagu) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java:17]: You should consider to inherit from AbstractExtension

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/VoltageExtensionSerializer.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
```
Comment (mathbagu) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/VoltageExtensionSerializer.java:2]: Change the copyright date.

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
```
Comment (mathbagu) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java:2]: Change the copyright date

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import com.powsybl.security.LimitViolation;
+
+/**
+ * Extension to handle pre-contingency voltage value for a voltage limit violation
+ *
+ * @author Olivier Bretteville <olivier.bretteville at rte-france.com>
+ */
+public class VoltageExtension extends AbstractExtension<LimitViolation> {
+
+    private final float preContingencyValue;
```
Comment (mathbagu) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java:18]: @sylvlecl Should we use a `float` or a `double` there?

**Comment by mathbagu** [comment]
@sylvlecl For `CurrentExtension` the security analysis API provides an interceptor that create this extension. Is it possible to do the same kind of interceptor for the `VoltageExtension`, or is it too specific to the simulator?

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java (hunk 4 above)
Comment (sylvlecl) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/extensions/VoltageExtension.java:18]: Ah, well spotted, we should use `double` as in the network and similar extensions.

**Comment by sylvlecl** [comment]
It would clearly be more consistent, but unfortunately we cannot implement it as long as we cannot find the `Bus` which corresponds to a given `LimitViolation`. Today, it's still the id of the `VoltageLevel` which is reported in the violation. 

So I think this could only be done after a deeper evolution about buses identification.

**Pull request merged**